src/ghdcbot/core/interfaces.py

```
@@ -74,6 +76,88 @@ def get_cursor(self, source: str) -> datetime | None:
+    def list_audit_events(self) -> list[dict]:
+        """Return all audit events."""
+
+    # Notifications
+
+    def was_notification_sent(self, dedupe_key: str) -> bool:
+        """Check if a notification was already sent (deduplication)."""
+
+    def mark_notification_sent(
+        self,
+        dedupe_key: str,
+        event: Any,
```

Ruff (0.15.7)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `event`

(ANN401)